src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptor.cs

```diff
@@ -20,35 +25,48 @@ internal TagHelperAttributeDescriptor(string name, PropertyInfo propertyInfo)
         /// Instantiates a new instance of the <see cref="TagHelperAttributeDescriptor"/> class.
         /// </summary>
         /// <param name="name">The HTML attribute name.</param>
-        /// <param name="propertyName">The name of the CLR property name that corresponds to the HTML
-        /// attribute.</param>
+        /// <param name="propertyName">The name of the CLR property that corresponds to the HTML attribute.</param>
         /// <param name="typeName">
-        /// The full name of the named (see <paramref name="propertyName"/>) property's
-        /// <see cref="System.Type"/>.
+        /// The full name of the named (see <paramref name="propertyName"/>) property's <see cref="System.Type"/>.
         /// </param>
-        public TagHelperAttributeDescriptor(string name,
-                                            string propertyName,
-                                            string typeName)
+        /// <param name="isStringProperty">
+        /// An indication whether this property is of type <see cref="string"/>.
+        /// </param>
+        public TagHelperAttributeDescriptor(
+            [NotNull] string name,
+            [NotNull] string propertyName,
+            [NotNull] string typeName,
+            bool isStringProperty)
```

Comment:
Optional: Remove this from the ctor params and have it do a check in the ctor body.